gitops/applications/base/monitoring-post-config/mimir-alertmanager.yaml

```diff
@@ -0,0 +1,109 @@
+apiVersion: redhatcop.redhat.io/v1alpha1
+kind: PasswordPolicy
+metadata:
+  name: alertmanager-bootstrap-password-policy
+  namespace: ${ARGOCD_ENV_vault_namespace}
+spec:
+  authentication:
+    path: ${ARGOCD_ENV_vault_k8s_admin_auth_path}
+    role: ${ARGOCD_ENV_vault_k8s_admin_role_name}
+    serviceAccount:
+      name: default
+  passwordPolicy: |
+    length = 5
+      rule "charset" {
+        charset = "-"
+        min-chars = 1
+      }
+---
+# NOTE: the secret generated by RandomSecret resources in this file are not used as is. 
```

Comment:
have you tested that the manually entered values do bot get overwritten by the random secret on a subsequent argocd sync operation?
overall I am confused why you define both random secrets and external secrets.

Reply:
> have you tested that the manually entered values do bot get overwritten by the random secret on a subsequent argocd sync operation?
@dfry yes. I have tested. I don't provide `refreshPeriod`, therefore random secret doesn't update the value after the creation of secret. However if we deleted the CR, then the value will be over-written in the vault. But we don't need to solve this problem. 

> overall I am confused why you define both random secrets and external secrets.
`RandomSecret` creates a secret in vault. That's it. It doesn't need to be random. But i don't have any easy way to make a static one either. 

`ExternalSecret` pulls the secret data from vault and formats alertmanager's configuration file.

Reply:
> have you tested that the manually entered values do bot get overwritten by the random secret on a subsequent argocd sync operation?

@dfry yes. I have tested. `refreshPeriod` is the field which forces value refresh. If we don't provide it, the secret is only created once.